lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2090 @@
+#!/usr/bin/python
+#
+# PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+# Copyright (C) 2016 PubNub Inc.
+# http://www.pubnub.com/
+# http://www.pubnub.com/terms
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'commiter',
+                    'version': '1.0'}
+
+
+DOCUMENTATION = '''
+---
+module: pubnub_blocks
+version_added: '2.2'
+short_description: PubNub blocks management module.
+description:
+  - This module allow to manage existing blocks by managing event handlers and starting/stopping block.
+author:
+  - PubNub <support@pubnub.com> (@pubnub)
+  - Sergey Mamontov <sergey@pubnub.com> (@parfeon)
+requirements:
+  - "python >= 2.7"
+options:
+  email:
+    description:
+      - Email from account for which new session should be started.
+      - Not required if C(cache) contains result of previous module call (in same play).
+    required: false
+  pwd:
+    description:
+      - Password which match to account to which specified C(email) belong.
+      - Not required if C(cache) contains result of previous module call (in same play).
+    required: false
+  cache:
+    description:
+      - In case if single play use blocks management module few times it is preferred to enabled 'caching' by making previous module to share gathered artifacts and pass them to this parameter.
+    required: false
+    default: {}
+  application:
+    description:
+      - Name of target PubNub application for which blocks configuration on specific C(keyset) will be done.
+    required: true
+  keyset:
+    description:
+      - Name of application's keys set which is bound to managed blocks.
+    required: true
+  state:
+    description:
+      - Intended block state after event handlers creation / update process will be completed.
+    required: false
+    default: 'start'
+    choices: ['start', 'stop', 'present', 'absent']
+  block:
+    description:
+      - Name of managed block which will be later visible on admin.pubnub.com.
+    required: true
+  description:
+    description:
+      - Short block description which will be later visible on admin.pubnub.com. Used only if block doesn't exists and won't change description for existing block.
+    required: false
+    default: 'New block'
+  event_handlers:
```

Comment:
Can these event handlers be specified in different tasks? For example:

```
- pubnub_block:
    name: foobar
    event_handlers:
      src: filename.js
      channels: namehere
- pubnub_block:
    name: foobar
    event_handlers:
      src: filename2.js
      channels: name2here
```

If that happened, what would the behavior be then? Will both handlers exist and be listening, or will only the second one work?